Assets/Packages/com.studio23.ss2.achievementsystem/Editor/AchievementDataWizard.cs

```diff
@@ -0,0 +1,132 @@
+			fontStyle = FontStyle.Bold,
+			fontSize = 18,
+			normal = new GUIStyleState
+			{
+				textColor = Color.cyan
+			}
+		};
+	}
+
+	private void OnGUI()
+	{
+		GUILayout.BeginScrollView(m_scrollPos);
+		GUILayout.Label("Achievement Data Creator", m_headerStyle);
+		GUILayout.Space(10f);
+
+        _achievementName = EditorGUILayout.TextField("Achievement Name", _achievementName);
+		GUILayout.Space(10f);
+
+		GUILayout.Label("Achievement Description");
+		GUILayout.BeginScrollView(m_descriptionScrollPosition);
+		_achievementDescription = EditorGUILayout.TextArea(_achievementDescription, GUILayout.Height(80));
+		GUILayout.EndScrollView();
+
+		_achievementSprite = (Sprite)EditorGUILayout.ObjectField("Achievement Icon", _achievementSprite, typeof(Sprite), false) as Sprite;
+		_achievementLockedSprite = (Sprite)EditorGUILayout.ObjectField("Achievement Locked Icon", _achievementLockedSprite, typeof(Sprite), false) as Sprite;
+
+		GUILayout.Space(10f);
+
+		GUILayout.Label("Achievement Type", EditorStyles.boldLabel);
+		_achievementType = (AchievementType)EditorGUILayout.EnumPopup(_achievementType);
+		if (_achievementType == AchievementType.ProgressTracked)
+		{
+			_achievementProgressGoal =
+				EditorGUILayout.FloatField("Achievement Progress Goal", _achievementProgressGoal);
+		}
+		GUILayout.FlexibleSpace();
+
+		if (GUILayout.Button("Create Achievement Data"))
+		{
+			CreateAchievementData();
+		}
+		GUILayout.EndScrollView();
+	}
+
+
+
+	private void CreateAchievementData()
+	{
+		if (IsFormIncomplete())
+			return;
+
+		AssetDatabase.Refresh();
+		AchievementData _data = ScriptableObject.CreateInstance<AchievementData>();
+
+		_data.AchievementName = _achievementName;
+		_data.AchievementDescription = _achievementDescription;
+		_data.LockedIcon = _achievementSprite.texture;
+		_data.UnlockedIcon = _achievementLockedSprite.texture;
+		_data.Type = _achievementType;
+		if (_achievementType == AchievementType.ProgressTracked)
+			_data.ProgressGoal = _achievementProgressGoal;
+		string path = $"Assets/Resources/AchievementSystem/AchievementData/{_achievementName}.asset";
+		AssetDatabase.CreateAsset(_data, path);
+		AssetDatabase.SaveAssets();
+		AssetDatabase.Refresh();
+		Debug.Log("Achievement data created and saved at " + path);
+	}
+
+	private bool IsFormIncomplete()
+	{
+		if (string.IsNullOrEmpty(_achievementName) || string.IsNullOrEmpty(_achievementDescription) ||
+			_achievementSprite == null || _achievementLockedSprite == null)
+		{
+			EditorUtility.DisplayDialog("Empty Fields", "Please ensure that all fields are filled.", "Ok");
+			return true;
+		}
+
+		if (File.Exists($"Assets/Resources/AchievementSystem/AchievementData/{_achievementName}.asset"))
+		{
+			bool result = EditorUtility.DisplayDialog("Already Exists",
+				"Achievement Data with the same name already exists!\nDo you want to overwrite it?", "Yes, replace.",
+				"No.");
+			if (!result)
+				return true;
+		}
+
+		return false;
+	}
```

Comment:
Over complicated. We are checking both IsFormInCompleted and IsFileExist at a time. We can break it down into two methods.  IsFormComplete() & IsFileExist():
```
private bool IsFormComplete()
{
    if (!string.IsNullOrEmpty(_achievementName) && !string.IsNullOrEmpty(_achievementDescription) &&
        _achievementSprite != null && _achievementLockedSprite != null &&
        !FileExists($"Assets/Resources/AchievementSystem/AchievementData/{_achievementName}.asset"))
    {
        return true;
    }

    ShowDialog("Incomplete Form", "Please fill in all fields and ensure unique achievement name.");
    return false;
}
```